### PR TITLE
Thchang/2046 improving filterable table widgets ux

### LINKED ISSUE #2046: [GUI UX] ambiguous button names in the catalog widget and the spectral line query widget
Improvements for the catalog widget to consider:
1. add the capability to trigger catalog filtering with a return in the filter fields.
2. rename the "Filter" button as "Apply filter" 
3. rename the "Reset" button as "Reset filter"
4. rename the "Close" button as "Close catalog" (this is to avoid confusion as "close widget")

Screen Shot 2022-11-17 at 13 53 12

Improvements for the spectral line query widget to consider:
1. add the capability to trigger catalog filtering with a return in the filter fields.
2. rename the "Filter" button as "Apply filter" 
3. rename the "Reset" button as "Reset filter"
4. rename the "Clear" button as "Clear plot"

Screen Shot 2022-11-17 at 13 58 28

### PR DESCRIPTION
**Pull request by TienHao**
**Description**

1. Closes #2046. 

2. Enabled applying filtering with enter/return in the filter fields.

3. Changed ambiguous text and colors of buttons in the catalog widget and the spectral line query widget. 
截圖 2023-07-25 下午4 23 35

**Checklist**

For linked issues (if there are):
- [x] assignee and label added
- [x] ZenHub issue connection, board status, and estimate updated

For the pull request:
- [x] reviewers and assignee added
- [x] ZenHub estimate, milestone, and release (if needed) added
- [x] ~e2e test passing~ / corresponding fix added
- [x] changelog updated / ~no changelog update needed~
- [x] ~protobuf updated to the latest dev commit~ / no protobuf update needed
- [x] `BackendService` unchanged / ~`BackendService` changed and corresponding ICD test fix added~

### CODE REVIEW

**Review by kswang1029 — APPROVED** [review]
looks good. a fix to xpath is applied to the relevant e2e tests in a branch.

File: src/components/CatalogOverlay/CatalogOverlayComponent.tsx [hunk 1]
```diff
@@ -832,9 +836,9 @@ export class CatalogOverlayComponent extends React.Component<WidgetProps> {
                     </div>
                     <div className="bp3-dialog-footer">
                         <div className="bp3-dialog-footer-actions">
-                            <AnchorButton intent={Intent.PRIMARY} text="Filter" onClick={this.handleFilterRequest} disabled={disable || !profileStore.updateTableView || !profileStore.hasFilter} />
-                            <AnchorButton intent={Intent.PRIMARY} text="Reset" onClick={this.handleResetClick} disabled={disable} />
-                            <AnchorButton intent={Intent.PRIMARY} text="Close" onClick={this.handleFileCloseClick} disabled={disable} />
+                            <AnchorButton intent={Intent.SUCCESS} text="Apply Filter" onClick={this.handleFilterRequest} disabled={disable || !profileStore.updateTableView || !profileStore.hasFilter} />
```
Comment (YuHsuan-Hwang) [inline src/components/CatalogOverlay/CatalogOverlayComponent.tsx]: Change the buttons to sentence case for consistency.

**Review by YuHsuan-Hwang — CHANGES_REQUESTED** [review]
The changes look good. 👍 Only one minor item:

**Review by crocka — APPROVED** [review]

File: src/components/CatalogOverlay/CatalogOverlayComponent.tsx (hunk 1 above)
Comment (TienHao) [reply src/components/CatalogOverlay/CatalogOverlayComponent.tsx]: thanks! just fixed.

**Review by YuHsuan-Hwang — APPROVED** [review]

**Pull request merged**